lib/model/folder.go

```diff
@@ -51,3 +51,7 @@ func (f *folder) scanSubdirsIfHealthy(subDirs []string) error {
 	}
 	return nil
 }
+
+func (f *folder) scanner() *folderScanner {
+	return &f.scan
```

Comment:
So I think this is worse.
I think what @calmh ment by encapsulation was doing
```
func (f *folder) Scan() {
   f.scan.Scan()
}
``` 
but I might be wrong.

Also, why just not embed scan into folder?